src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,1931 @@
+              IL_001b:  ret
+            }
+            """);
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Opposite_Implicit(LanguageVersion langVersion)
+    {
+        var source = """
+            class C
+            {
+                 int[] M(System.Span<int> arg) => arg;
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (3,39): error CS0029: Cannot implicitly convert type 'System.Span<int>' to 'int[]'
+            //      int[] M(System.Span<int> arg) => arg;
+            Diagnostic(ErrorCode.ERR_NoImplicitConv, "arg").WithArguments("System.Span<int>", "int[]").WithLocation(3, 39));
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Opposite_Explicit(LanguageVersion langVersion)
+    {
+        var source = """
+            class C
+            {
+                 int[] M(System.Span<int> arg) => (int[])arg;
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (3,39): error CS0030: Cannot convert type 'System.Span<int>' to 'int[]'
+            //      int[] M(System.Span<int> arg) => (int[])arg;
+            Diagnostic(ErrorCode.ERR_NoExplicitConv, "(int[])arg").WithArguments("System.Span<int>", "int[]").WithLocation(3, 39));
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Opposite_Explicit_UserDefined(LanguageVersion langVersion)
+    {
+        var source = """
+            class C
+            {
+                 int[] M(System.Span<int> arg) => (int[])arg;
+            }
+
+            namespace System
+            {
+                readonly ref struct Span<T>
+                {
+                    public static explicit operator T[](Span<T> span) => throw null;
+                }
+            }
+            """;
+        var verifier = CompileAndVerify(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion));
+        verifier.VerifyDiagnostics();
+        verifier.VerifyIL("C.M", """
+            {
+              // Code size        7 (0x7)
+              .maxstack  1
+              IL_0000:  ldarg.1
+              IL_0001:  call       "int[] System.Span<int>.op_Explicit(System.Span<int>)"
+              IL_0006:  ret
+            }
+            """);
+    }
+
+    [Fact]
+    public void Conversion_Array_Span_Implicit_Params()
+    {
+        var source = """
+            using System;
+
+            class C
+            {
+                void M(string[] a)
+                {
+                    M1(a);
+                    M2(a);
+                }
+                void M1(params Span<string> s) { }
+                void M2(params ReadOnlySpan<string> s) { }
+            }
```

Comment:
>} [](http://example.com/codeflow?start=12&length=1)

Perhaps test with:
```csharp
void M(string[] a)
{
    M1(a);
    M2(a);
}
void M1(params object[] a) { }
void M2(params ReadOnlySpan<object> s) { }

```

Reply:
Are we testing this case?

Reply:
Ah, I probably tested Span<object> instead of the suggested object[]. I will extend the test below, thanks.